File: code/rust/217_contains_duplicate.rs
use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut seen = HashSet::with_capacity(nums.len());
        for &n in &nums {
            if !seen.insert(n) {
                return true;
            }
        }
        false
    }
}

File: code/python/217_contains_duplicate.py
from typing import List


class Solution:
    """
    LeetCode 217. Contains Duplicate
    https://leetcode.com/problems/contains-duplicate/
    """

    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False